quiz/src/test/java/rx/quiz/combining/IntermediateCombineTimersKoanTest.java

```diff
@@ -0,0 +1,24 @@
+package rx.quiz.combining;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableTransformer;
+import io.reactivex.Scheduler;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class IntermediateCombineTimersKoanTest extends IntermediateCombineTimersKoan {
+
+    @NotNull
+    @Override
+    protected ObservableTransformer<String, String> timeoutIfFirstElementDoesNotArriveIn(
+            @NotNull Scheduler scheduler, long delay, @NotNull TimeUnit unit) {
+        return upstream -> {
+            Observable<String> timeout = Observable.create(emitter ->
+                    scheduler.scheduleDirect(() -> emitter.onError(new TimeoutException()), delay, unit));
+
+            return upstream.ambWith(timeout);
```

Comment:
Nice! amb is the right operator for this.

You can also make it a little shorter: Observable<String> timeout = Observable.never().timeout(delay,unit, scheduler);

Reply:
Looks shorter. It's a very interesting use of Observable.never().